src/backend/distributed/utils/multi_partitioning_utils.c

```diff
@@ -555,54 +547,50 @@ CreateFixPartitionShardIndexNamesTaskList(Oid parentRelationId, Oid partitionRel
 	LockShardListMetadata(parentShardIntervalList, ShareLock);
 
 	int taskId = 1;
-	List *taskList = NIL;
 
 	ShardInterval *parentShardInterval = NULL;
 	foreach_ptr(parentShardInterval, parentShardIntervalList)
 	{
+		MemoryContext localContext = AllocSetContextCreate(CurrentMemoryContext,
+														   "CreateFixPartitionShardIndexNames",
+														   ALLOCSET_DEFAULT_SIZES);
+		MemoryContext oldContext = MemoryContextSwitchTo(localContext);
+
 		uint64 parentShardId = parentShardInterval->shardId;
 
 		List *queryStringList =
 			WorkerFixPartitionShardIndexNamesCommandList(parentShardId,
 														 parentIndexIdList,
 														 partitionRelationId);
-
 		if (queryStringList != NIL)
 		{
 			Task *task = CitusMakeNode(Task);
 			task->jobId = INVALID_JOB_ID;
 			task->taskId = taskId++;
-
 			task->taskType = DDL_TASK;
 
-			/*
-			 * There could be O(#partitions * #indexes) queries in
-			 * the queryStringList.
-			 *
-			 * In order to avoid round-trips per query in queryStringList,
-			 * we join the string and send as a single command via the UDF.
-			 * Otherwise, the executor sends each command with one
-			 * round-trip.
-			 */
-			char *string = StringJoin(queryStringList, ';');
-			StringInfo commandToRun = makeStringInfo();
-
-			appendStringInfo(commandToRun,
-							 "SELECT pg_catalog.citus_run_local_command($$%s$$)", string);
-			SetTaskQueryString(task, commandToRun->data);
+			char *prefix = "SELECT pg_catalog.citus_run_local_command($$";
+			char *postfix = "$$)";
+			char *string = StringJoinParams(queryStringList, ';', prefix, postfix);
+
+			SetTaskQueryString(task, string);
+
 
 			task->dependentTaskList = NULL;
 			task->replicationModel = REPLICATION_MODEL_INVALID;
 			task->anchorShardId = parentShardId;
 			task->taskPlacementList = ActiveShardPlacementList(parentShardId);
 
-			taskList = lappend(taskList, task);
+			bool localExecutionSupported = true;
+			ExecuteUtilityTaskList(list_make1(task), localExecutionSupported);
```

Comment:
now we are regressing cross node parallelism, but still all my tests shows that this performs better. Still, looking for a way to run at least 1 task per node.

But seems non-trivial